peppy/project.py

```diff
@@ -140,16 +150,146 @@ def __init__(
+
+        dict_self["_samples"] = [s.to_dict() for s in self.samples]
+        dict_other["_samples"] = [s.to_dict() for s in other.samples]
+
+        return dict_self == dict_other
+
+    def _convert_to_dict(self, project_value=None):
+        """
+        Recursively transform various project values, objects, and attributes to a dictionary
+        compatible format. Useful for creating an extended dictionary representation
+        of the peppy project.
+
+        :param project_value object - the value to transform
+        """
+        if isinstance(project_value, list):
+            new_list = []
+            for item_value in project_value:
+                new_list.append(self._convert_to_dict(item_value))
+            return new_list
+
+        elif isinstance(project_value, dict):
+            new_dict = {}
+            for key, value in project_value.items():
+                if key != "_project":
+                    new_dict[key] = self._convert_to_dict(value)
+            return new_dict
+
+        elif isinstance(project_value, PathExAttMap):
+            new_dict = PathExAttMap.to_dict(project_value)
+            return self._convert_to_dict(new_dict)
+            # return new_dict
+
+        elif isinstance(project_value, Sample):
+            new_dict = PathExAttMap.to_dict(project_value)
+            return new_dict
+
+        elif isinstance(project_value, pd.DataFrame):
+            project_value = project_value.to_dict()
+            project_value = self._nan_converter(project_value)
+            return project_value
+
+        else:
+            return project_value
+
+    def _nan_converter(self, nan_dict: dict) -> dict or list:
+        """
+        Searching and converting nan values to None
+        :param dict nan_dict: dictionary with nan values
+        """
+        if isinstance(nan_dict, list):
+            new_list = []
+            for list_item in nan_dict:
+                new_list.append(self._nan_converter(list_item))
+
+            return new_list
+
+        elif isinstance(nan_dict, dict):
+            new_dict = {}
+            for key, value in nan_dict.items():
+                new_dict[key] = self._nan_converter(value)
+            return new_dict
+        elif isinstance(nan_dict, float):
+            if math.isnan(nan_dict):
+                return None
+        else:
+            return nan_dict
+
+    def from_dict(self, d: dict) -> None:
+        """
+        Init a peppy project instance from a dictionary representation
+        of an already processed PEP.
+        :param dict d: in-memory dict representation of processed pep.
+        """
+        _LOGGER.info(f"Processing project from dictionary...")
+        if CONFIG_KEY not in self:
+            self[CONFIG_KEY] = PathExAttMap()
+
+        # extract custom index for sample table if exists
+        self.st_index = (
+            d[SAMPLE_TABLE_INDEX_KEY] if SAMPLE_TABLE_INDEX_KEY in d else None
+        )
```

Comment:
Just another example where you can replace a comment with helper function :grinning: :

 ```suggestion
        self.st_index = self.extract_custom_index_for_sample()
        
        (...)
        def extract_custom_index_for_sample(self, pep_dictionary: Dict):
            return pep_dictionary[SAMPLE_TABLE_INDEX_KEY] if SAMPLE_TABLE_INDEX_KEY in pep_dictionary else None
        )
```